packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/OpenShiftVirtualMachinesRow.tsx

```diff
@@ -9,6 +11,11 @@ import { VMCellProps, VmData, VMNameCellRenderer } from './components';
 const cellRenderers: Record<string, React.FC<VMCellProps>> = {
   name: VMNameCellRenderer,
   status: PowerStateCellRenderer,
+  template: ({ data }) => (
+    <TableCell>
+      {(data?.vm as OpenshiftVM)?.object?.metadata?.labels?.['vm.kubevirt.io/template'] ?? ''}
+    </TableCell>
```

Comment:
> we should not need to cast, this is an Openshift specific method, it should already get the correct type.
>

As already discussed, this is not happening because `VMCellProps` is not generic.

> this is a long renderer, it dos not look right implemented inline, i see two options, one is implementing a method that return the template string, another is implementing a cell rendered and using i
>

The renderer is very simple, it's the the property path that is too long:)  In the previous approach ("flat" objects) this was handled in the flattening method. In the current approach ("blob" objects) we need to dive deep to get the prop.  Btw  in the k8s world this seems a common pattern - adding a method for each nested prop would is not scalable.

Reply:
> lets consider a getVmTemplate(vm: ProviderVirtualMachine) if will check the type, and return the currect 
>

Done. Note that:
1. right now it works only for OpenShift
2. the same value will be retrieved in 2 different ways (jsonpath and accessor) - this impacts maintenance. 
3. following this pattern (accessor method per prop) will result in many accessor methods. In the long term we need a structured way to handle them i.e. folder structure and an utility file per type. As for now I've created a `vmProps.ts` utility.